doc/decisions/operation_sequences.md

```diff
@@ -0,0 +1,253 @@
+# Operation sequences
+
+## Problem
+
+The current public API allows various kind of sequences of `kdbGet` and `kdbSet` calls.
+There is no real guideline of allowed and/or prohibited sequences, other than `kdbGet` for a certain parent key must be called before `kdbSet`:
+
+- initially after `kdbOpen`
+- after conflict errors in `kdbSet`
+
+We discovered that some sequences are currently legal but problematic for change tracking and maybe also other use cases.
+
+Currently, notification plugins assume that a `kdbSet` operation for a certain parent key always directly follows the `kdbGet` operation for that same key.
```

Comment:
```suggestion
Currently, notification plugins assume that a `kdbSet` operation for a certain parent key always follows the `kdbGet` operation for that same key, without a different `kdbGet` in between.
```

"directly" suggests `kdbGet (...); kdbSet (...);` without _any code_ in between, which obviously doesn't do anything.